lib/delegate/index.js

```diff
@@ -26,8 +26,28 @@ var Iterator = require('./iterator');
 exports.createHandler = function (registries) {
 
     return function (req, reply) {
+        var good, start;
         var iter = new Iterator(registries, factory);
 
+        if (good = req.server.plugins.good) {
+            start = Date.now();
+            req.raw.res.on('timeout', function () {
```

Comment:
Bound on the response which has a finite lifetime, and the destroy the socket immediately afterwards which will prevent the timeout event on the response from happening again.

That said, might as well change to once.